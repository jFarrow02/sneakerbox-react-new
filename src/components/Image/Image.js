import React from 'react';
import './Image.scss';
import propTypes from 'prop-types';


const Image = props=>{
    const {src, alt, width, height, isLink, linkUrl, title} = props;
    const anchor = isLink ? <a href={linkUrl}></a> : null;
    return(
        <img
            alt={alt}
            height={height}
            src={src}
            title={title}
            width={width}
        >
            {anchor}
        </img>
    )
}

export default Image;

Image.propTypes = {
    alt: propTypes.string,
    height: propTypes.oneOf([propTypes.string, propTypes.number]),
    isLink: propTypes.bool,
    linkUrl: propTypes.string,
    src: propTypes.string,
    title: propTypes.oneOf([propTypes.string, propTypes.number]),
    width: propTypes.oneOf([propTypes.string, propTypes.number]),
};

Image.defaultProps = {
    alt: '',
    height: '',
    isLink: false,
    linkUrl: '',
    src: '',
    title: '',
    width: '',
}