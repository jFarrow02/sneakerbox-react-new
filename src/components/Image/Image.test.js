import React from 'react';
import Image from './Image';
import renderer from 'react-test-renderer';

it('should match a snapshot', ()=>{
    const tree = renderer.create(<Image/>)
        .toJSON();
    expect(tree).toMatchSnapshot();
})