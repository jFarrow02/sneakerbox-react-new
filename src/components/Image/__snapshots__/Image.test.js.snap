// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match a snapshot 1`] = `
<img
  alt=""
  height=""
  src=""
  title=""
  width=""
/>
`;
